src/main/java/org/wso2/carbon/inbound/poll/dbeventlistener/DBEventPollingConsumer.java

```diff
@@ -0,0 +1,380 @@
+            if (isRollback(msgCtx)) {
+                return false;
+            } else {
+                if (filteringCriteria.equals(DBEventConstants.DB_FILTERING_BY_TIMESTAMP)) {
+                    dbEventListnerRegistryHandler.writeToRegistry(registryPath, lastProcessedTimestamp);
+                }
+                if(!isConnectionAlive()) {
+                    createConnection();
+                }
+                if (StringUtils.isNotEmpty(deleteQuery)) {
+                    statement = connection.prepareStatement(deleteQuery);
+                    query = deleteQuery;
+                    statement.execute(deleteQuery);
+                } else if (StringUtils.isNotEmpty(updateQuery)) {
+                    statement = connection.prepareStatement(updateQuery);
+                    query = updateQuery;
+                    statement.execute(updateQuery);
+                }
+            }
+        } catch (SQLException e) {
+            log.error("Error while capturing the change data " + query, e);
+        }
+        return true;
+    }
+
+    private MessageContext createMessageContext() {
+        MessageContext msgCtx = synapseEnvironment.createMessageContext();
+        org.apache.axis2.context.MessageContext axis2MsgCtx = ((org.apache.synapse.core.axis2.Axis2MessageContext) msgCtx)
+                .getAxis2MessageContext();
+        axis2MsgCtx.setServerSide(true);
+        axis2MsgCtx.setMessageID(UUIDGenerator.getUUID());
+        msgCtx.setProperty(org.apache.axis2.context.MessageContext.CLIENT_API_NON_BLOCKING, true);
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        axis2MsgCtx.setProperty(MultitenantConstants.TENANT_DOMAIN, carbonContext.getTenantDomain());
+        return msgCtx;
+    }
+
+    /**
+     * Execute the query to retrieve the records, create each record as OMElement and inject to the sequence
+     */
+    private void fetchDataAndInject() {
+        Statement statement = null;
+        ResultSet rs = null;
+        String tableName = properties.getProperty(DBEventConstants.DB_TABLE);
+        DBEventRegistryHandler dbEventListnerRegistryHandler = new DBEventRegistryHandler();
+        String lastUpdatedTimestampFromRegistry = null;
+        if (filteringCriteria.equals(DBEventConstants.DB_FILTERING_BY_TIMESTAMP)) {
+            lastUpdatedTimestampFromRegistry = dbEventListnerRegistryHandler.readFromRegistry(registryPath).toString();
+        }
+        String dbScript = buildQuery(tableName, filteringCriteria,
+                filteringColumnName, lastUpdatedTimestampFromRegistry);
+        try {
+            if(!isConnectionAlive()) {
+                createConnection();
+            }
+            statement = connection.prepareStatement(dbScript);
+            rs = statement.executeQuery(dbScript);
+            ResultSetMetaData metaData = rs.getMetaData();
+            List<String> primaryKeys = Arrays.asList(primaryKeysFromConfig);
```

Comment:
Well it may be a edge case :) and I don't think we need to support it for the moment.